src/contracts/core/DelegationManager.sol

```diff
@@ -38,14 +38,14 @@ contract DelegationManager is
     // @dev Chain ID at the time of contract deployment
     uint256 internal immutable ORIGINAL_CHAIN_ID;
 
-    // @dev Maximum Value for `stakerOptOutWindowBlocks`. Approximately equivalent to 6 months in blocks.
-    uint256 public constant MAX_STAKER_OPT_OUT_WINDOW_BLOCKS = (180 days) / 12;
+    /// @dev The minimum number of blocks to complete a withdrawal of a strategy. 50400 * 12 seconds = 1 week
+    uint256 public constant LEGACY_MIN_WITHDRAWAL_DELAY_BLOCKS = 50400;
 
-    // The max configurable withdrawal delay per strategy. Set to 30 days in seconds
-    uint256 public constant MAX_WITHDRAWAL_DELAY = 30 days;
+    /// @dev Wed Jan 01 2025 17:00:00 GMT+0000, timestamp used to check whether a pending withdrawal
+    /// should be processed as legacy M2 or with slashing considered.
+    uint32 public constant LEGACY_WITHDRAWALS_TIMESTAMP = 1735750800;
 
-    // the number of 12-second blocks in 30 days (60 * 60 * 24 * 30 / 12 = 216,000)
-    uint256 public constant MAX_WITHDRAWAL_DELAY_BLOCKS = MAX_WITHDRAWAL_DELAY / 12;
+    uint32 public constant WITHDRAWAL_DELAY = SlashingConstants.DEALLOCATION_DELAY;
```

Comment:
This should not be in a library, this is problematic for different envs. it should be an immutable

Reply:
Then I guess same applies for DEALLOCATION_DELAY in the AllocationManager

Reply:
yeah. not sure the best solution is here.

I feel like it's immutables